pennylane/capture/primitives.py

```diff
@@ -117,3 +202,122 @@ def _(*_, **__):
+
+    @primitive.def_abstract_eval
+    def _(*_, **__):
+        abstract_eval = measurement_type._abstract_eval  # pylint: disable=protected-access
+        return abstract_type(abstract_eval, n_wires=None)
+
+    return primitive
+
+
+def create_measurement_mcm_primitive(
+    measurement_type: Type["qml.measurements.MeasurementProcess"], name: str
+) -> Optional["jax.core.Primitive"]:
+    """Create a primitive corresponding to the input type where the abstract inputs are classical
+    mid circuit measurement results.
+
+    Called by default when defining any class inheriting from :class:`~.MeasurementProcess`, and is used to
+    set the ``MeasurementProcesss._mcm_primitive`` property.
+
+    Args:
+        measurement_type (type): a subclass of :class:`~.MeasurementProcess`
+        name (str): the preferred string name for the class. For example, ``"expval"``.
+            ``"_mcm"`` is appended to this name for the name of the primitive.
+
+    Returns:
+        Optional[jax.core.Primitive]: A new jax primitive. ``None`` is returned if jax is not available.
+    """
+
+    if not has_jax:
+        return None
+
+    primitive = jax.core.Primitive(name + "_mcm")
+
+    @primitive.def_impl
+    def _(*mcms, **kwargs):
+        raise NotImplementedError(
+            "mcm measurements do not currently have a concrete implementation"
+        )
+        # need to figure out how to convert a jaxpr int into a measurement value, and pass
+        # that measurment value here.
+        # return type.__call__(measurement_type, obs=mcms, **kwargs)
+
+    abstract_type = _get_abstract_measurement()
+
+    @primitive.def_abstract_eval
+    def _(*mcms, **__):
+        abstract_eval = measurement_type._abstract_eval  # pylint: disable=protected-access
+        return abstract_type(abstract_eval, n_wires=len(mcms))
+
+    return primitive
+
+
+def create_measurement_wires_primitive(
+    measurement_type: type, name: str
+) -> Optional["jax.core.Primitive"]:
+    """Create a primitive corresponding to the input type where the abstract inputs are the wires.
+
+    Called by default when defining any class inheriting from :class:`~.MeasurementProcess`, and is used to
+    set the ``MeasurementProcesss._wires_primitive`` property.
+
+    Args:
+        measurement_type (type): a subclass of :class:`~.MeasurementProcess`
+        name (str): the preferred string name for the class. For example, ``"expval"``.
+            ``"_wires"`` is appended to this name for the name of the primitive.
+
+    Returns:
+        Optional[jax.core.Primitive]: A new jax primitive. ``None`` is returned if jax is not available.
+    """
+    if not has_jax:
+        return None
+
+    primitive = jax.core.Primitive(name + "_wires")
+
+    @primitive.def_impl
+    def _(*args, has_eigvals=False, **kwargs):
+        if has_eigvals:
+            wires = args[:-1]
+            kwargs["eigvals"] = args[-1]
+        else:
+            wires = qml.wires.Wires(args)
```

Comment:
Why do we have the explicit casting to `Wires` without eigvals but not with eigvals? :thinking: